env.go

```diff
@@ -19,47 +59,32 @@ func (e *Env) Run(steward Steward) error {
 		return err
 	}
 
-	usageHint := ""
-	unsetVar := ""
-	setVar := ""
-	quoteReplacement := "\""
-	switch e.Shell {
-	case "fish":
-		usageHint = "# To load these variables into your shell, execute:\n#   eval (%s)\n"
-		unsetVar = "set -e %s;\n"
-		setVar = "set -x %s \"%s\";\n"
-		quoteReplacement = "\\\""
-	default:
-		usageHint = "# To load these variables into your shell, execute:\n#   eval $(%s)\n"
-		unsetVar = "unset %s\n"
-		setVar = "export %s=\"%s\"\n"
-		quoteReplacement = "\\\""
+	var templateStr string
+	format := e.Format
+
+	if format == "shell" {
+		format = e.DetectedShell
 	}
 
-	vars := env.Variables()
+	if foundTemplate, ok := envFormatters[format]; ok {
+		templateStr = foundTemplate
+	} else {
+		templateStr = format
+	}
+	tmpl, err := template.New("envTmpl").Funcs(templateFuncMap).Parse(templateStr)
 
-	// sort the vars to unset
-	sort.Strings(vars.Unset)
+	vals := templateVals{}
+	variables := env.Variables()
 
-	// sort the vars to set
-	var keys []string
-	for key := range vars.Set {
-		keys = append(keys, key)
-	}
-	sort.Strings(keys)
+	vals.Set = variables.Set
 
-	// display the vars using the format string for the shell
-	if e.UsageHint {
-		fmt.Printf(usageHint, strings.Join(os.Args, " "))
-	}
+	sort.Strings(variables.Unset)
+	vals.Unset = variables.Unset
 
-	for _, key := range vars.Unset {
-		fmt.Printf(unsetVar, key)
-	}
+	vals.Command = strings.Join(os.Args, " ")
```

Comment:
I believe you had a previous patchset that passed the `Command` into the `Env` object, that seems more correct to me. So, I think we should add `Command` back to the `Env` struct and move the `strings.Join(os.Args, " ")` out to the arg parser (where I should have put it to begin with…).